docs/aws/audit/cloudfrontmonitoring/rules/cloudfront_distributions_https_remediation.mdx

```diff
@@ -1,6 +1,260 @@
+   ```
+
+By following these steps, you can ensure that HTTPS is enabled on your CloudFront distributions using a Python script.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the CloudFront service. You can find this by typing 'CloudFront' into the search bar at the top of the console.
+3. In the CloudFront console, you will see a list of your CloudFront distributions. Click on the ID of the distribution you want to check.
+4. In the distribution settings, click on the 'Behaviors' tab. Here, you can see the 'Viewer Protocol Policy'. If it is set to 'HTTP and HTTPS', 'Redirect HTTP to HTTPS', or 'HTTPS Only', then HTTPS is enabled. If it is set to 'HTTP Only', then HTTPS is not enabled.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   Installation:
+   ```
+   pip install awscli
+   ```
+   Configuration:
+   ```
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all CloudFront distributions: Use the following AWS CLI command to list all the CloudFront distributions in your AWS account:
+
+   ```
+   aws cloudfront list-distributions --query 'DistributionList.Items[*].Id' --output text
+   ```
+   This command will return a list of IDs for all your CloudFront distributions.
+
+3. Check HTTPS configuration for each distribution: For each distribution ID returned by the previous command, run the following command to check its HTTPS configuration:
+
+   ```
+   aws cloudfront get-distribution-config --id <distribution-id>
+   ```
+   Replace `<distribution-id>` with the actual ID of your CloudFront distribution. This command will return the configuration details of the specified distribution.
+
+4. Check the ViewerProtocolPolicy: In the output of the previous command, look for the `ViewerProtocolPolicy` field under `DefaultCacheBehavior`. If the value of this field is `allow-all`, it means that both HTTP and HTTPS are enabled on the distribution. If the value is `https-only`, it means that only HTTPS is enabled.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3). You will need to set up your AWS credentials and AWS region to use Boto3. You can install Boto3 using pip:
+
+```python
+pip install boto3
+```
+
+2. Import the necessary modules and create a session using your credentials:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>